<a href="https://colab.research.google.com/github/purnawanpp/plants/blob/main/plants.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!wget https://github.com/purnawanpp/plants/raw/main/plants
!wget https://github.com/purnawanpp/plants/raw/main/spores
!apt install openbabel
!chmod +x /content/plants
!chmod +x /content/spores

--2023-03-08 04:58:59--  https://github.com/purnawanpp/plants/raw/main/plants
Resolving github.com (github.com)... 140.82.113.4
Connecting to github.com (github.com)|140.82.113.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/purnawanpp/plants/main/plants [following]
--2023-03-08 04:59:00--  https://raw.githubusercontent.com/purnawanpp/plants/main/plants
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9536847 (9.1M) [application/octet-stream]
Saving to: ‘plants’

plants              100%[===================>]   9.09M  --.-KB/s    in 0.1s    

2023-03-08 04:59:00 (91.8 MB/s) - ‘plants’ saved [9536847/9536847]

--2023-03-08 04:59:00--  https://github.com/purnawanpp/plants/raw/main/spores
Resolvi

Download Protein PDB ID 3HTB

In [2]:
!wget https://files.rcsb.org/download/3HTB.pdb

--2023-03-08 04:59:18--  https://files.rcsb.org/download/3HTB.pdb
Resolving files.rcsb.org (files.rcsb.org)... 128.6.158.70
Connecting to files.rcsb.org (files.rcsb.org)|128.6.158.70|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘3HTB.pdb’

3HTB.pdb                [ <=>                ] 161.13K  --.-KB/s    in 0.08s   

2023-03-08 04:59:18 (1.89 MB/s) - ‘3HTB.pdb’ saved [164997]



Separete Protein and ligand

In [3]:
!./spores --mode splitpdb 3HTB.pdb


				 SPORES        

	      Structure PrOtonation and REcognition System

			      version 1.3


author: T. ten Brink

scientific contributors: P. Duchstein, J. Gimmler, O. Korb, M. Zerara,
			  T.E. Exner

contact: tim.tenbrink@uni-konstanz.de


molecule number: 1
64 alternative atomic positions have been removed !
splitting 3HTB
225
protein: 0
ligs: 4
water: 660
metal: 0


**Add Hydrogen, charge (gastaiger) dan protonasi pada pH 7.4**

Keterangan:

1. -p 7.4: menetapkan pH 7.4 untuk menghitung muatan.
2. -b: menghitung muatan terbentuk pada pH yang ditetapkan.
3. --ff "gasteiger": menetapkan metode perhitungan muatan menggunakan metode Gasteiger.

In [16]:
!obabel -imol2 /content/protein.mol2 -omol2 -h -p 7.4 -b --ff "gasteiger" -O rec.mol2
!obabel -imol2 /content/ligand_JZ4167_0.mol2 -omol2 -h -p 7.4 -b --ff "gasteiger" -O lig.mol2

*** Open Babel Warning  in ReadMolecule
  Failed to kekulize aromatic bonds in MOL2 file (title is 3HTB_protein)

*** Open Babel Warning  in DoTransformations
  Both -p and -h options are set. All implicit hydrogens (-h) will be added without considering pH.

1 molecule converted
*** Open Babel Warning  in DoTransformations
  Both -p and -h options are set. All implicit hydrogens (-h) will be added without considering pH.

1 molecule converted


Membuat Binding Sitenya atau Gridboxnya

In [17]:
!./plants --mode bind lig.mol2 rec.mol2



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile



Calculated bindingsite definition (from 1 ligands): 

bindingsite_center 22.7142 -25.2627 -3.283
bindingsite_radius 3.75624



Pindahkan data:

*   bindingsite_center 22.7142 -25.2627 -3.283
*   bindingsite_radius 3.75624

kedalam file **dock_gcolab.txt**

Simulasi Molecular Docking

In [19]:
!./plants --mode screen /content/dock_gcolab.txt



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile

PLANTS warning: Config file parser found unknown token '' in line 2.
PLANTS warning: Please check config file line 2. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token '' in line 6.
PLANTS warning: Please check config file line 6. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token '' in line 10.
PLANTS warning: Please check config file line 10. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token 'flexible_protein_side_chain_11' in line 12.
PLANTS warning: Config file parser found unknown token '' in line 13.
PLANTS warning: Please check config file line 13. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token '' i

Perhitungan RMSD hasil simulasi

In [20]:
!obrms -f lig.mol2 /content/results/3HTB_ligand_2_entry_00001_conf_01.mol2

RMSD 3HTB_ligand_2:3HTB_ligand_2_entry_00001_conf_01 0.449554
